src/langchain_google_firestore/vectorstores.py

```diff
@@ -0,0 +1,354 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Any, Callable, Iterable, List, Optional, Type
+
+import more_itertools
+import numpy as np
+from google.cloud import firestore  # type: ignore
+from google.cloud.firestore import (  # type: ignore
+    Client,
+    CollectionReference,
+    DocumentSnapshot,
+)
+from google.cloud.firestore_v1.base_query import BaseFilter  # type: ignore
+from google.cloud.firestore_v1.base_vector_query import DistanceMeasure  # type: ignore
+from google.cloud.firestore_v1.vector import Vector  # type: ignore
+from langchain_community.vectorstores.utils import maximal_marginal_relevance
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from langchain_core.vectorstores import VectorStore
+
+from langchain_google_firestore.document_converter import convert_firestore_document
+
+from .version import __version__
+
+USER_AGENT = "langchain-google-firestore-python:vectorstore" + __version__
+WRITE_BATCH_SIZE = 500
+DEFAULT_TOP_K = 4
+
+
+class FirestoreVectorStore(VectorStore):
+    """Interface for vector store."""
+
+    def __init__(
+        self,
+        collection: CollectionReference | str,
+        embedding: Embeddings,
```

Comment:
for consistency across products can we use `embedding_service` as the variable name